### ISSUE TITLE
HookedRootModule.add_caching_hooks() doesn't use model device, fails if model on e.g. cuda:1 device

### ISSUE THREAD

**Issue by montemac** [neelnanda-io/TransformerLens#202]
From line 233 of hook_points.py:
```
if device is None:
    device = "cuda" if torch.cuda.is_available() else "cpu"
```

It seems logical to place cached activations on the model's device from `self.cfg.model` if set, to avoid needing to constantly pass around `cuda:1` to these calls in cases where multiple GPUs are present?

(I have the one-line fix in my fork and can make a PR unless anyone objects.)

**Comment by jbloomAus** [comment]
Sounds very reasonable to me. I'll accept the PR. I have a vague goal of understanding how to test this stuff in the future but for extant bugs with simple fixes, I'd prefer to accept PRs.

**Comment by jbloomAus** [comment]
This will be sorted out when we solve https://github.com/neelnanda-io/TransformerLens/issues/246.

**Comment by jbloomAus** [comment]
solved by #250
